wp-content/themes/sample-theme/blocks/src/core--button/block.json

```diff
@@ -0,0 +1,4 @@
+{
+    "script": "file:script.js",
```

Comment:
This `block.json` is used only as an entry point to `wp-scripts build` logic here https://github.com/WordPress/gutenberg/blob/4f53c6b5c4d21df1c29689adeb45e8c5732ab724/packages/scripts/config/webpack.config.js#L108-L121